vertical-pod-autoscaler/pkg/recommender/input/metrics/metrics_client.go

```diff
@@ -48,20 +48,23 @@ type MetricsClient interface {
 
 type metricsClient struct {
 	metricsGetter resourceclient.PodMetricsesGetter
+	namespace     string
 }
 
 // NewMetricsClient creates new instance of MetricsClient, which is used by recommender.
 // It requires an instance of PodMetricsesGetter, which is used for underlying communication with metrics server.
-func NewMetricsClient(metricsGetter resourceclient.PodMetricsesGetter) MetricsClient {
+// namespace limits queries to particular namespace, empty strings means all namespaces.
```

Comment:
nit: I think it's better to say, use k8sapiv1.NamespaceAll for all namespaces (it's unlikely to change from the empty string, but still)